lib/eex/lib/eex/engine.ex

```diff
@@ -66,13 +66,23 @@ defmodule EEx.Engine do
   """
   def handle_assign({:@, meta, [{name, _, atom}]}) when is_atom(name) and is_atom(atom) do
     line = meta[:line] || 0
-    quote line: line, do: Dict.get(var!(assigns), unquote(name))
+    quote line: line, do: EEx.Engine.fetch_assign!(var!(assigns), unquote(name))
   end
-
   def handle_assign(arg) do
     arg
   end
 
+  @doc false
+  def fetch_assign!(assigns, key) do
+    case Dict.fetch(assigns, key) do
+      :error ->
+        raise ArgumentError, message: """
+        assign @#{key} not available in eex template. Available assigns: #{inspect Dict.keys(assigns)}
+        """
```

Comment:
This will create a newline after the error message, use a normal string instead.